api/provider/aws/environment_create.go

```diff
@@ -81,39 +49,69 @@ func (e *EnvironmentProvider) Create(req models.CreateEnvironmentRequest) (strin
 		return "", err
 	}
 
-	launchConfigName := fqEnvironmentID
-	if err := e.createLC(
-		launchConfigName,
-		aws.StringValue(securityGroup.GroupId),
-		instanceType,
-		e.Config.InstanceProfile(),
-		e.Config.SSHKeyPair(),
-		amiID,
-		userData); err != nil {
-		return "", err
-	}
+	// creating asg, lc isn't required for dynamic environments
+	if strings.ToLower(req.EnvironmentType) != models.EnvironmentTypeDynamic {
```

Comment:
[nit] best practice is to use "positive" bool comparisons. It's also the more conservative path in this instance (e.g. if there were a third type of environment, we wouldn't necessarily want this block to execute).

Reply:
Not convinced about "positive" comparisons being best practice.

I actually ended up changing this from `== static` to `== dynamic` with the justification that existing behavior is the default behavior. Any new environment types should use the existing as the baseline. But I can see the argument the other way too.

Reply:
I can't find where I read this, but the argument goes something like this:

Obviously context matters, but when you have an either-or situation, it's better to use positive comparisons because that's how humans naturally communicate. For example, it's more natural to say "If Seshi is a male, then he has XY chromosomes" than it is to say "If Seshi is not a female, then he has XY chromosomes".